lib/resolve-script/glob-patterns.js

```diff
@@ -1,14 +1,31 @@
 var path = require('path')
+var resolvePkg = require('resolve-pkg')
 
-module.exports = function (dir1, dir2) {
+module.exports = function (dir1, dir2, options) {
   var expanded = path.resolve(dir1, path.join.apply(this, dir2.split(':')))
 
-  return [
+  var globArray = [
     // exact file match (+ any same-named extensions)
     expanded + '+(|.*)',
     // a nested index file match (+ any same-named extensions)
     path.join(expanded, 'index+(|.*)'),
     // any nested files at all
     path.join(expanded, '*')
   ]
+
+  if (options && Array.isArray(options.modules)) {
+    options.modules.forEach(function (moduleName) {
+      const modulePath = resolvePkg(moduleName + '/scripts')
+
+      if (modulePath) {
+        var moduleLocation = path.resolve(modulePath, path.join.apply(this, dir2.split(':')))
+
+        globArray.push(moduleLocation + '+(|.*)')
+        globArray.push(path.join(moduleLocation, 'index+(|.*)'))
+        globArray.push(path.join(moduleLocation, '*'))
+      }
+    })
+  }
```

Comment:
This method was pretty tidy previously and now has a 2-nested branch condition shoehorned into it. Could we extract this to a well named private function? Maybe:

```js
return [
  stuff,
  that, 
  was, 
  here
].concat(arrayOfNewNestedStuff(options.modules))
```